pkg/provision/providers/vm/disk.go

```diff
@@ -11,20 +11,52 @@ import (
 	"github.com/talos-systems/talos/pkg/provision"
 )
 
-// CreateDisk creates an empty disk file.
-func (p *Provisioner) CreateDisk(state *State, nodeReq provision.NodeRequest) (diskPath string, err error) {
-	diskPath = state.GetRelativePath(fmt.Sprintf("%s.disk", nodeReq.Name))
+// UserDiskName returns disk device path.
+func (p *Provisioner) UserDiskName(index int) string {
+	res := "/dev/vd"
 
-	var diskF *os.File
+	var convert func(i int) string
 
-	diskF, err = os.Create(diskPath)
-	if err != nil {
-		return
+	convert = func(i int) string {
+		remainder := i % 26
+		divider := i / 26
+
+		prefix := ""
+
+		if divider != 0 {
+			prefix = convert(divider - 1)
+		}
+
+		return fmt.Sprintf("%s%s", prefix, string(rune('a'+remainder)))
```

Comment:
This could probably use a comment on how it works. Its not obvious, at least for me. Where does the number 26 come from?

Reply:
oh yeah. It's all how disk naming works:
```
vda
vdb
vdc
...
vdz
vdaa
vdab
...
```

Reply:
26 is amount of letters in the English alphabet.

Reply:
Ah, ok I see it now.